classes/shaman.lua

```diff
@@ -107,19 +128,48 @@ local function checkRollingThunderRuneAndLightningSieldStacks(self, ...)
         return;
     end
 
-    local RollingThunderEquipped = C_Engraving and SAO:IsSpellLearned(432056);
+    local RollingThunderEquipped = (C_Engraving and SAO:IsSpellLearned(432056)) or (SAO:GetTalentByName((GetSpellInfo(88766)))) ~= 0;
     if not RollingThunderEquipped then
         RollingThunderHandler:deactivate();
     else
-        -- C_UnitAuras is currently available for Classic Era only
-        -- Fortunately, RollingThunderHandler is used only on Season of Discovery
+        -- C_UnitAuras is currently available for Classic Era and Cataclysm only
+        -- Fortunately, RollingThunderHandler is used only on Season of Discovery and Cataclysm
         local aura = C_UnitAuras.GetAuraDataBySpellName("player", GetSpellInfo(324));
-        if aura and aura.applications > 6 and not SAO:GetActiveOverlay(324) then
-            RollingThunderHandler:activate(aura.applications);
+        if SAO:IsCata() then
+            if aura and aura.applications > 5 and not SAO:GetActiveOverlay(324) then
+                RollingThunderHandler:activate(aura.applications);
+            end
+        else
+            if aura and aura.applications > 6 and not SAO:GetActiveOverlay(324) then
+                RollingThunderHandler:activate(aura.applications);
+            end
         end
     end
 end
 
+local function rollingThunderCombatCheck(combat)
+    local TimetoLingerGlow = 2.5; -- Buttons glows temporarily for 2.5 secs
+    if not combat then RollingThunderHandler.glowtimer = C_Timer.NewTimer(
+        TimetoLingerGlow,
+        function() RollingThunderHandler:deactivate(); end
+    )
+    end
+    if combat then
+        if RollingThunderHandler.glowtimer then
+            RollingThunderHandler.glowtimer:Cancel();

```

Comment:
For extra safety, you might want to set the glowtimer to nil after cancel, for example
```lua
RollingThunderHandler.glowtimer:Cancel();
RollingThunderHandler.glowtimer = nil;
```